File: react-image-exchange-main/src/types/planogram.ts
export interface PlanogramSection {
  column: number;
  expected_product: string;
  allowed_variants: string[];
  min_quantity: number;
  max_quantity: number;
}

export interface PlanogramShelf {
  row: number;
  sections: PlanogramSection[];
}

export interface PlanogramCreate {
  name: string;
  shelves: PlanogramShelf[];
}

export interface Planogram extends PlanogramCreate {
  id: string;
  created_at?: string;
  updated_at?: string;
}

export interface ComplianceIssue {
  row: number;
  column: number;
  issue_type: string;
  expected: string;
  found: string;
  severity: "high" | "medium" | "low";
}

export interface ComplianceResult {
  is_compliant: boolean;
  compliance_score: number;
  issues: ComplianceIssue[];
  correct_placements: number;
  total_positions: number;
}

File: image-recognition-api/app/models/planogram.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class PlanogramSection(BaseModel):
    column: int
    expected_product: str
    allowed_variants: List[str]
    min_quantity: int = 1
    max_quantity: int = 1

class PlanogramShelf(BaseModel):
    row: int
    sections: List[PlanogramSection]

class PlanogramCreate(BaseModel):
    name: str
    description: Optional[str] = None
    shelves: List[PlanogramShelf]

class Planogram(PlanogramCreate):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None 